src/iceberg/CMakeLists.txt

```diff
@@ -17,11 +17,16 @@
 
 set(ICEBERG_SOURCES demo_table.cc)
 
+set(ICEBERG_AVRO_STATIC_BUILD_LIBS)
+list(APPEND ICEBERG_AVRO_STATIC_BUILD_LIBS Avro::avro_static)
+
 add_iceberg_lib(iceberg
                 SOURCES
                 ${ICEBERG_SOURCES}
                 PRIVATE_INCLUDES
-                ${ICEBERG_INCLUDES})
+                ${ICEBERG_INCLUDES}
+                STATIC_LINK_LIBS
```

Comment:
Should we follow `libiceberg-arrow` to prefer `libavrocpp_s` for static linking and `libavrocpp` for dynamic linking?